### PR TITLE
fixup missing job selfLink in api response

### PR DESCRIPTION
**Pull request by ErikZigo**
Fixes : https://github.com/keboola/google-bigquery-extractor/issues/38
Fixes: https://github.com/keboola/google-bigquery-extractor/issues/37

### CODE REVIEW

File: src/Keboola/Google/BigQuery/RestApi/Job.php [hunk 1]
```diff
@@ -120,6 +120,26 @@ private function validateStatus(Response $response, Client $client)
         return true;
     }
 
+    private function getJobUrl(array $responseBody)
+    {
+        if (empty($responseBody['jobReference']['projectId'] || $responseBody['jobReference']['jobId'])) {
```
Comment (ujovlado) [inline src/Keboola/Google/BigQuery/RestApi/Job.php]: ~Nerozpisal by si to radsej na `&&`? ... akoze `empty(x) && empty(y)`~

**Comment by ujovlado** [comment]
@ErikZigo ako je to teda?  Chodi tam stale `projectId` a obcas sa stane, ze nepride `jobId`?

File: src/Keboola/Google/BigQuery/RestApi/Job.php [hunk 2]
```diff
@@ -120,6 +120,26 @@ private function validateStatus(Response $response, Client $client)
...
+            $this->logger->error(sprintf('%s: Google API error - missing job reference', $this->name));
+            throw new \Exception('Google API error - missing job reference');
+        }
+
+        $url = sprintf(self::API_ENDPOINT, $responseBody['jobReference']['projectId']);
+        unset($responseBody['jobReference']['projectId']);
+
+        $url .= '/' . $responseBody['jobReference']['jobId'];
```
Comment (ujovlado) [inline src/Keboola/Google/BigQuery/RestApi/Job.php:133]: nebude toto robit problem, ked nebudem mat `jobId`?

File: src/Keboola/Google/BigQuery/RestApi/Job.php (hunk 1 above)
Comment (ErikZigo) [reply src/Keboola/Google/BigQuery/RestApi/Job.php]: jj ma to byt `&&`

File: src/Keboola/Google/BigQuery/RestApi/Job.php (hunk 2 above)
Comment (ujovlado) [reply src/Keboola/Google/BigQuery/RestApi/Job.php:133]: aha, tam slo o `selfLink`, ok

**Comment by ujovlado** [comment]
Ok, uz vidim. Myslel som si, ze tam obcas chyba `jobId`, ked bola zmienka o jobe. Ale je to ten `selfLink`.

**Comment by ujovlado** [comment]
Tak toto ignoruj: https://github.com/keboola/google-bigquery-extractor/pull/40#issuecomment-461775115

File: src/Keboola/Google/BigQuery/RestApi/Job.php (hunk 1 above)
Comment (ujovlado) [reply src/Keboola/Google/BigQuery/RestApi/Job.php]: Ok, tak to je jedine, co mi prislo divne.

Comment (ujovlado) [reply src/Keboola/Google/BigQuery/RestApi/Job.php]: Ale taram, sorry. Ma to byt `empty(x) || empty(y)`

Comment (ujovlado) [reply src/Keboola/Google/BigQuery/RestApi/Job.php]: Len sa mi nezda, ze je to v kope :)

Comment (ErikZigo) [reply src/Keboola/Google/BigQuery/RestApi/Job.php]: jj vubec nechapu jak jsme tu podminku vytvoril

**Comment by ErikZigo** [comment]
jeste jsem prihodil fix issue co jsi daval k sync akci a internal erroru

**Review by ujovlado — APPROVED** [review]
cajk

**Pull request merged**